v2/spiffetls/mode.go

```diff
@@ -27,7 +32,8 @@ type DialMode interface {
 }
 
 type dialMode struct {
-	tlsType    authMode
+	mode       clientMode
+	raw        bool
```

Comment:
nit: for someone looking at this for the first time this may not be super obvious, we may add a comment explaining what is this.

Reply:
i renamed the variable and added a comment.